### PR TITLE
fix: expose ShowInSearch field in access UI

### PR DESCRIPTION
**Pull request by wilr**
Upgrading from SilverStripe 3 to SilverStripe 5 it appears the ShowInSearch field is no longer displayed in the UI. 

The File record still contains this data and the implementation was to match Page instances which contain the 'Show in search' option to CMS editors.

## Issue
- https://github.com/silverstripe/silverstripe-asset-admin/issues/1400

### CODE REVIEW

**Comment by ChloeSartorelli** [comment]
There's something odd happening with the behat tests... can you please rebase your PR to see if that resolves it?

**Comment by wilr** [comment]
Rebased now.

**Comment by ChloeSartorelli** [comment]
This PR doesn't seem to fix it - I still can't see the ShowInSearch field when editing a file in the asset admin:
image

Can you please advise if there's some additional steps I need to take, or if I'm looking in the wrong place, etc?

**Comment by wilr** [comment]
@GuySartorelli should be under 'Permissions'

**Comment by ChloeSartorelli** [comment]
Oh, interesting. Was it under "permissions" in CMS 3? That doesn't seem like an intuitive place for it, but if that's where it was in CMS 3 then I'd be okay with it.

Please add an exception for Folder - there is a comment in the $db definition that says it shouldn't be inherited for folder, so we shouldn't show the field for folders.

**Comment by wilr** [comment]
@GuySartorelli Done added a check to exclude folder.

**Comment by ChloeSartorelli** [comment]
Awesome, thanks for making that change. Can you please also answer the question from my previous comment?
> Was it under "permissions" in CMS 3? That doesn't seem like an intuitive place for it, but if that's where it was in CMS 3 then I'd be okay with it.

File: code/Forms/AssetFormFactory.php [hunk 1]
```diff
@@ -152,13 +153,13 @@ protected function getSaveAction($record)
     {
         if ($record && $record->isInDB() && $record->canEdit()) {
             /** @var FormAction $action */
-            $action = FormAction::create('save', _t(__CLASS__.'.SAVE', 'Save'))
+            $action = FormAction::create('save', _t(__CLASS__ . '.SAVE', 'Save'))
```
Comment (ChloeSartorelli) [inline code/Forms/AssetFormFactory.php:156]: Can you please either avoid making unrelated changes like this (and raise a separate PR for them) or at least do these in a separate commit? It is very hard to see which parts of the PR are relevant changes vs just small nothing-changes like this when you have them all in one commit.

Comment (wilr) [reply code/Forms/AssetFormFactory.php:156]: @GuySartorelli Pro-tip just chuck a `?w=1` on the URL and you can see the changes sans the Vscode auto-linting fix. Ideally we should have some sort of hooks so things are linted by default when coming in.

Comment (ChloeSartorelli) [reply code/Forms/AssetFormFactory.php:156]: Thanks for the tip - my request still stands for future pull requests (you can leave as is for this one). I don't like ignoring anything when reviewing pull requests.

**Comment by ChloeSartorelli** [comment]
Can you please resolve the behat failures, and also respond to my question in the previous comments?

**Comment by wilr** [comment]
> Can you please resolve the behat failures, and also respond to my question in the previous comments?

Not sure about 3 but I think it's fine under Permissions as it relates to some form of Permission (is it permitted to be in search).

**Comment by ChloeSartorelli** [comment]
Permissions is more about _user_ permissions than what the record is "permitted" to do. I wouldn't like to make a precedent that this sort of thing is a "permission".

If this is where it was for CMS 3 then I'm fine with it. If it wasn't there in CMS 3 or we can't verify one way or another, please move it out of permissions.

**Comment by ChloeSartorelli** [comment]
This pull request hasn't had any activity for a while. If you're still interested in having this merged, can you please either move it out of permissions or else validate where it was in CMS 3? If there's no more action after a few weeks I will close the PR, and you can reopen it when you're ready to work on it again.

**Comment by wilr** [comment]
Hi @GuySartorelli happy to move it out to main. In this 3 build there were no tabs. Everything is under 'Main'.

**Comment by ChloeSartorelli** [comment]
Sounds good - when that change is made, I'll be happy to re-review and merge.

**Comment by wilr** [comment]
@GuySartorelli done.

**Review by ChloeSartorelli — APPROVED** [review]
LGTM, will merge on green.
